driver-core/src/main/com/mongodb/operation/CommandOperationHelper.java

```diff
@@ -712,6 +716,18 @@ static boolean shouldAttemptToRetry(final boolean retryWritesEnabled, final Thro
         return retryWritesEnabled && isRetryableException(exception);
     }
 
+    static void logRetryExecute(final String operation, final Throwable originalError) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(format("Retrying operation %s due to an error \"%s\"", operation, originalError));
+        }
+    }
+
+    static void logUnableToRetry(final String operation, final Throwable originalError) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(format("Unable to retry operation %s due to error \"%s\"", operation, originalError));
```

Comment:
It appears like this log message will also be generate when the driver could retry, but doesn't because retryWrites is disabled. If that's the case, then it's misleading to say that the driver is "unable to retry".  

I don't think the driver should log anything if retryWrites is disabled.  It should only log if retryWrites is enabled but is unable to retry.

Reply:
The calls to `logUnableToRetry` in `MixedBulkWriteOperation` are only executed if `retryWrites` is  `true`. The one call in `CommandOperationHelper` at line 458 needs an extra check to ensure that `retryWrites` is enabled.

Reply:
I have changed the calls in `CommandOperationHelper` to check if `retryWrites` is enabled:
```
                    if (!shouldAttemptToRetry(command, e)) {
                        if (isRetryWritesEnabled(command)) {
                            logUnableToRetry(command.getFirstKey(), e);
                        }
                        <process exception>
                    }
```